### ISSUE TITLE
Checkinstall package

### ISSUE THREAD

**Issue by JL2210** [skycocker/chromebrew#2602]
I could create a checkinstall package, which would make it much easier to package crew packages. They use roughly the same format, and they catch files not installed by DESTDIR.

**Comment by uberhacker** [comment]
Sounds interesting.

**Comment by JL2210** [comment]
Here's a copy. See if it works. 
```ruby
require 'package'

class Checkinstall < Package
  description 'CheckInstall is a computer program for Unix-like operating systems which eases the installation and uninstallation of software compiled from source by making use of package management systems.'
  homepage 'http://asic-linux.com.mx/~izto/checkinstall/'
  version '1.6.2'
  source_url 'http://asic-linux.com.mx/~izto/checkinstall/files/source/checkinstall-1.6.2.tar.gz'
  source_sha256 'dc61192cf7b8286d42c44abae6cf594ee52eafc08bfad0bea9d434b73dd593f4'

  binary_url ({
  })
  binary_sha256 ({
  })

  def self.patch
    system 'wget', 'https://github.com/JL2210/patches/raw/da54bdfb4aea5dfbab8c08224ebd9416fc2bc62d/checkinstall.patch'
    abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('checkinstall.patch') ) == '71c36071df8deaf3c14caa58d79fd86eb9262872281226c2ee0e48b640c182ce'
    system 'patch -p1 < checkinstall.patch'
  end

  def self.build
    system 'make'
  end

  def self.install
    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/bin/"
    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/sbin/"
    FileUtils.mkdir_p "#{CREW_DEST_LIB_PREFIX}/"
    system "make", "install", "PREFIX=#{CREW_PREFIX}", "LCDIR=#{CREW_LIB_PREFIX}/checkinstall", "CONFDIR=#{CREW_LIB_PREFIX}/checkinstall/locale", "LIBDIR=#{CREW_LIB_PREFIX}"
    system "cp -pa /usr/local/bin/installwatch #{CREW_DEST_PREFIX}/bin/"
    system "cp -rpa /usr/local/lib64/checkinstall/ #{CREW_DEST_LIB_PREFIX}/"
    system "cp -pa /usr/local/lib64/installwatch.so #{CREW_DEST_LIB_PREFIX}/"
    system "cp -pa /usr/local/sbin/checkinstall #{CREW_DEST_PREFIX}/sbin/"
    system "cp -pa /usr/local/sbin/makepak #{CREW_DEST_PREFIX}/sbin/"
  end
end
```

**Comment by uberhacker** [comment]
Make sure you change `/usr/local` -> `#{CREW_PREFIX}` and `/usr/local/lib64` -> `#{CREW_LIB_PREFIX}` or I cannot install on my arm or i686 VM. :)

**Comment by uberhacker** [comment]
It works but this is a bit easier:
```
require 'package'

class Checkinstall < Package
  description 'CheckInstall is a computer program for Unix-like operating systems which eases the installation and uninstallation of software compiled from source by making use of package management systems.'
  homepage 'http://asic-linux.com.mx/~izto/checkinstall/'
  version '1.6.2'
  source_url 'http://asic-linux.com.mx/~izto/checkinstall/files/source/checkinstall-1.6.2.tar.gz'
  source_sha256 'dc61192cf7b8286d42c44abae6cf594ee52eafc08bfad0bea9d434b73dd593f4'

  binary_url ({
  })
  binary_sha256 ({
  })

  def self.patch
    system 'wget', 'https://github.com/JL2210/patches/raw/da54bdfb4aea5dfbab8c08224ebd9416fc2bc62d/checkinstall.patch'
    abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('checkinstall.patch') ) == '71c36071df8deaf3c14caa58d79fd86eb9262872281226c2ee0e48b640c182ce'
    system 'patch -p1 < checkinstall.patch'
  end

  def self.build
    system 'make'
  end

  def self.install
    system "make",
           "install",
           "PREFIX=#{CREW_DEST_PREFIX}",
           "LIBDIR=#{CREW_DEST_LIB_PREFIX}",
           "LCDIR=#{CREW_DEST_LIB_PREFIX}/checkinstall",
           "CONFDIR=#{CREW_DEST_PREFIX}/etc/checkinstall"
    system "install -Dm644 #{CREW_DEST_PREFIX}/checkinstallrc #{CREW_DEST_HOME}/.checkinstallrc"
    system "cp #{CREW_DEST_PREFIX}/checkinstallrc $HOME/.checkinstallrc"
  end

  def self.postinstall
    puts
    puts "To configure, modify $HOME/.checkinstallrc".lightblue
    puts
  end
end
```
Not sure this is usable in Chromebrew, however.
```
Please choose the packaging method you want to use.
Slackware [S], RPM [R] or Debian [D]?
```
We want non-distro specific tarball packaging. :)

**Comment by JL2210** [comment]
Slackware.

**Comment by JL2210** [comment]
Check the patch. I'm not sure it's the best way.

**Comment by uberhacker** [comment]
Dude, it works and we don't want Slackware packaging, we want Chromebrew packaging.  There is quite a bit of difference.

**Comment by JL2210** [comment]
We could always use `installwatch`, from the same package.

**Comment by uberhacker** [comment]
Fixed in #2647.
